metricbeat/module/ceph/health/data.go

```diff
@@ -0,0 +1,138 @@
+package health
+
+import (
+	"encoding/json"
+	"io"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type Tick struct {
+	time.Time
+}
+
+var format = "2006-01-02 15:04:05"
+
+func (t *Tick) MarshalJSON() ([]byte, error) {
+	return []byte(t.Time.Format(format)), nil
+}
+
+func (t *Tick) UnmarshalJSON(b []byte) (err error) {
+	b = b[1 : len(b)-1]
+	t.Time, err = time.Parse(format, string(b))
+	return
+}
+
+type StoreStats struct {
+	BytesTotal  int64  `json:"bytes_total"`
+	BytesLog    int64  `json:"bytes_log"`
+	LastUpdated string `json:"last_updated"`
+	BytesMisc   int64  `json:"bytes_misc"`
+	BytesSSt    int64  `json:"bytes_sst"`
+}
+
+type Mon struct {
+	LastUpdated  Tick       `json:"last_updated"`
+	Name         string     `json:"name"`
+	AvailPercent int64      `json:"avail_percent"`
+	KbTotal      int64      `json:"kb_total"`
+	KbAvail      int64      `json:"kb_avail"`
+	Health       string     `json:"health"`
+	KbUsed       int64      `json:"kb_used"`
+	StoreStats   StoreStats `json:"store_stats"`
+}
+
+type HealthServices struct {
+	Mons []Mon `json:"mons"`
+}
+
+type Health struct {
+	HealthServices []HealthServices `json:"health_services"`
+}
+
+type Timecheck struct {
+	RoundStatus string `json:"round_status"`
+	Epoch       int64  `json:"epoch"`
+	Round       int64  `json:"round"`
+}
+
+type Output struct {
+	OverallStatus string    `json:"overall_status"`
+	Timechecks    Timecheck `json:"timechecks"`
+	Health        Health    `json:"health"`
+}
+
+type HealthRequest struct {
+	Status string `json:"status"`
+	Output Output `json:"output"`
+}
+
+func eventsMapping(body io.Reader) []common.MapStr {
+
+	var d HealthRequest
+	err := json.NewDecoder(body).Decode(&d)
+	if err != nil {
+		logp.Err("Error: ", err)
+	}
+
+	events := []common.MapStr{}
+
+	event := common.MapStr{
+		"cluster": common.MapStr{
+			"overall_stats": d.Output.OverallStatus,
+			"timechecks": common.MapStr{
+				"epoch": d.Output.Timechecks.Epoch,
+				"round": common.MapStr{
+					"value":  d.Output.Timechecks.Round,
+					"status": d.Output.Timechecks.RoundStatus,
+				},
+			},
+		},
+	}
+
+	events = append(events, event)
+
+	for _, HealthService := range d.Output.Health.HealthServices {
```

Comment:
@amandahla I just realised now that we have a problem here. We send two different event types as part of the same metricset. We need to extract this part here into its own metricset. Not should war we should call it. `monitor`?